Core/Src/main.c

```diff
@@ -143,7 +156,17 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t read_adc() {
+	uint8_t adc_tx_data[3] = {0x01, 0x80, 0x00}; // transmission data based on ADC data sheet
+	uint8_t adc_rx_data[3] = {0}; // empty array to store received data from ADC
 
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); // set CS to LOW to initiate communication
+    HAL_SPI_TransmitReceive(&hspi1, adc_tx_data, adc_rx_data, 3, HAL_MAX_DELAY); // transmit and receive data over SPI protocol
```

Comment:
Why use the max delay here? If it does fail, it will just stall here and your program wouldn't do anything about it... please use an appropriate timeout such as 200ms.

Also since you have encapsulated this functionality in a function (which is good!) what happens if this function fails? What value will your function return? I am not entirely too sure... maybe it will not write anything to the buffer in which case you will return 0, which is definitely incorrect.

There is a reason that all these HAL functions return a status code, it is so you can be sure if the function failed or not, and if it has failed, you can take the appropriate action.

In this case, the best thing to do would likely be to RETURN the status code and instead of having this function take no parameters, you could pass a uint16_t pointer and populate it with the adc value. This way, in main, you can use the status code to determine if you want to use the adc value or not.